src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,159 @@
+        newUser.setLogin("valid login");
+        expectedMessage = "User password can`t be null";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                expectedMessage,
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user login length less then 6 characters")
+    void register_UserLoginLengthLessThenSixCharacters_notOk() {
+        newUser.setLogin("test");
+        newUser.setPassword("valid password");
+        expectedMessage = "User login %s length less then %d characters";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                String.format(expectedMessage, newUser.getLogin(), 6),
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user password length less then 6 characters")
+    void register_UserPasswordLengthLessThenSixCharacters_notOk() {
+        newUser.setLogin("valid login");
+        newUser.setPassword("test");
+        expectedMessage = "User password %s length less then %d characters";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                String.format(expectedMessage, newUser.getPassword(), 6),
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user age less then 18")
+    void register_UserAgeLessThenEighteen_notOk() {
+        newUser.setLogin("valid login");
+        newUser.setPassword("valid password");
+        newUser.setAge(17);
+        expectedMessage = "User age %d less then %d years";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                String.format(expectedMessage, newUser.getAge(), 18),
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user with such login already exists")
+    void register_UserAlreadyExists_notOk() {
+        User oldUser = new User(1L, "valid login", "valid password", 18);
+        storageDao.add(oldUser);
+        newUser = new User(2L, "valid login", "other valid password", 18);
```

Comment:
unregistered users do not have ID yet
```suggestion
        newUser = new User("valid login", "other valid password", 18);
```